### DISCUSSION TITLE
kernel/riscv32/handler.S のSPPビットに関するコメントについて

### DISCUSSION THREAD

**Discussion by yak1ex** [nuta/microkernel-book#23] [Corrections]
### ページ番号 (図・表番号)

p.101 リスト7.1
ソースコード上は
https://github.com/nuta/microkernel-book/blob/c46c2de86f3c917eefc2b86f2a440b4b80c0fefa/kernel/riscv32/handler.S#L14

### 概要

```bash
書籍を使用した勉強会 https://jisaku-micro-kernel.connpass.com/event/333487/ に参加しているのですがその中で疑問が生じましたので確認させてください。RISC-Vについて知識がほとんどなく誤りであれば申し訳ありません。

   andi a0, a0, (1 << 8)      // SPPビット: 0ならカーネルモードで発生。
```
と記載されていますがSPPビットが1のときにスーパバイザモード＝カーネルモードではないでしょうか？
同様の判定と思われる
https://github.com/nuta/microkernel-book/blob/c46c2de86f3c917eefc2b86f2a440b4b80c0fefa/kernel/riscv32/trap.c#L99
においても0の時に「ユーザーモードで発生した」との記載があります。
```

**Comment by nuta** [comment]
こんにちは。ご指摘の通り、`handler.S`のコメントは「 SPPビット: **1** ならカーネルモードで発生。」が正しい挙動です。
正誤表に追加されるよう対応しておきます。ご報告ありがとうございます。
